feature/search/api/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/search/api/navigation/SearchNavigation.kt

```diff
@@ -14,17 +14,24 @@
  * limitations under the License.
  */
 
-package com.google.samples.apps.nowinandroid.feature.search.navigation
+package com.google.samples.apps.nowinandroid.feature.search.api.navigation
 
 import androidx.navigation.NavController
 import androidx.navigation.NavGraphBuilder
 import androidx.navigation.NavOptions
-import androidx.navigation.compose.composable
-import com.google.samples.apps.nowinandroid.feature.search.SearchRoute
+import com.google.samples.apps.nowinandroid.core.navigation.NiaBackStack
 import kotlinx.serialization.Serializable
 
 @Serializable data object SearchRoute
 
+@Serializable data class SearchRouteNav3(val onInterestsClick: () -> Unit)
+
+fun NiaBackStack.navigateToSearch(
+    onInterestsClick: () -> Unit,
```

Comment:
There shouldn't be any need to pass this. We can just call `NiaBackStack::navigateToTopic` directly.

Reply:
Yup updated.